pkg/ccl/backupccl/restore_span_covering.go

```diff
@@ -182,45 +200,72 @@ func makeSimpleImportSpans(
 						break
 					}
 					f := it.Value()
-					fspan := endKeyInclusiveSpan(f.Span)
-					if sp := span.Intersect(fspan); sp.Valid() {
-						fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
-						if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
-							fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
-						}
+					fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
+					if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
+						fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
+					}
 
-						// Lookup the size of the file being added; if the backup didn't
-						// record a file size, just assume it is 16mb for estimating.
-						sz := f.EntryCounts.DataSize
-						if sz == 0 {
-							sz = 16 << 20
+					// Lookup the size of the file being added; if the backup didn't
+					// record a file size, just assume it is 16mb for estimating.
+					sz := f.EntryCounts.DataSize
+					if sz == 0 {
+						sz = 16 << 20
+					}
+
+					if inclusiveOverlap(span, f.Span) {
+						var startKey roachpb.Key
+						var endKey roachpb.Key
+						if sp := span.Intersect(f.Span); sp.Valid() {
+							startKey = sp.Key
+							endKey = sp.EndKey
+						} else {
+							startKey = f.Span.EndKey
+							endKey = f.Span.EndKey
 						}
 
 						if len(cover) == spanCoverStart {
```

Comment:
could you add a comment here that describes when this branch is taken? I think it's "while evaluating the given filtered span, we have yet to add a new restore span entry to the cover"

Reply:
in addition, i'd write a sentence explaining the decision within the branch on when to drain vs add: the add branch is taken bc of case 1 in your `endKeyNotCoveredFiles` description.

Reply:
er, actually I think this is case 2 as you describe above.

Reply:
Yeah it's example 2. I added some additional comments just to guide the reader, and moved the place where the covering span entry is created so it's more inline with the other place that creates the entry.

Reply:
thanks for adding more clarity here!